code-monkey/src/instrumentation/__init__.py

```diff
@@ -64,11 +72,41 @@ def initialize_tracer(attributes: Attributes = None):
     set_tracer(trace.get_tracer("replayio.ai-playground"))
 
 
-def instrument(name: str, attributes=None):
+def instrument(name: str, params: List[str] = None, attributes: Attributes = None):
     def decorator(func):
         @functools.wraps(func)
         def wrapper(*args, **kwargs):
```

Comment:
Function inside a function inside a function... or... when decorators decorate decorators! :thinking:

Reply:
yeah, I think if the decorator didn't take parameters itself, it'd only be function inside a function.  alas :)